### DISCUSSION TITLE
Cannot get HyperHDR to recognise HyperSerialESP32 on S2-Mini

### DISCUSSION THREAD

**Discussion by mattsmithuk** [awawa-dev/HyperSerialESP32#37] [General]
Hi

I have been unable to get HyperHDR to recognise my S2-Mini.

On the logs I get the line

`2023-06-19T16:31:50.026Z [LEDDEVICE_ADALIGHT] Device disabled, device 'adalight' signals error: 'Permission denied'`

I have serial connections on both /dev/ttyS0 and /dev/ttyS1 and neither work
The user running HyperHDR is part of both the tty and dialout groups

I have tried flashing the S2-mini with the baudrate set to 5000000 and 2000000 and neither work

My platformio.ini configuration is here: (https://raw.githubusercontent.com/mattsmithuk/HyperSerialESP32/master/platformio.ini) - compiled using the GitHub Action

My HyperHDR log is here: (https://pastebin.com/hzVuzSVD)

Any thoughts?

**Comment by awawa-dev** [comment]
Hi
Neither of your devices in your OS is esp32-s2 mini lolin and all return invalid/unknown VID: 0x0, PID: 0x0 (usually some system devices or malfunction driver). Should be PID: 0x80c2 and VID 0x303a for s2-mini.

**Comment by awawa-dev** [comment]
Maybe your esp32-s2 came clean without bootloader so it doesnt work at all? Try to flash factory version of the firmware from your build artifacts at the correct flash address (manual on the main page) and check if other pc/laptop will detect esp32-s2 serial port after that (in the system devices, HyperHDR is not necessary for such test).
